src/odemis/driver/xt_client.py

```diff
@@ -2047,19 +2047,29 @@ def _getPosition(self):
         pos = self.parent.get_stage_position()
         pos["rx"] = pos.pop("t")
         pos["rz"] = pos.pop("r")
+        # Make sure the full rotations are within the range (because the SEM
+        # actually allows rotation in both directions)
+        for an in ("rx", "rz"):
+            rng = self.axes[an].range
+            if util.almost_equal(rng[1] - rng[0], 2 * math.pi):
+                pos[an] = pos[an] % (2 * math.pi) - rng[0]
```

Comment:
are you sure this is correct? For example, if `pos[an] = 0.5*math.pi` and the range is `[-math.pi, +math.pi]` I think this line will update `pos[an]` to `1.5*math.pi` instead of staying `0.5*math.pi`

Reply:
Indeed! I fixed it to `(pos[an] - rng[0]) % (2 * math.pi) + rng[0]`. It looks to me like the proper formula, right?

Reply:
yes that looks correct